cloud/terraform/templates.go

```diff
@@ -305,14 +306,32 @@ echo $PRIVATEIP > /tmp/.ip
 	"PUBLICIP=`ip route get 8.8.8.8 | awk '{printf \"%s\", $NF; exit}'`" + `
 
 
-kubeadm init --apiserver-bind-port ${PORT} --token ${TOKEN} --kubernetes-version v${CONTROL_PLANE_VERSION} \
-             --apiserver-advertise-address ${PUBLICIP} --apiserver-cert-extra-sans ${PUBLICIP} ${PRIVATEIP} \
-             --service-cidr ${SERVICE_CIDR}
-
-# install weavenet
-sysctl net.bridge.bridge-nf-call-iptables=1
-export kubever=$(kubectl version --kubeconfig /etc/kubernetes/admin.conf | base64 | tr -d '\n')
-kubectl apply --kubeconfig /etc/kubernetes/admin.conf -f "https://cloud.weave.works/k8s/net?env.CHECKPOINT_DISABLE=1&env.IPALLOC_RANGE=${POD_CIDR}&disable-npc=true&k8s-version=$kubever"
+KUBEADM_FLAGS=" --apiserver-bind-port ${PORT}"
+KUBEADM_FLAGS+=" --token ${TOKEN}"
+KUBEADM_FLAGS+=" --kubernetes-version v${CONTROL_PLANE_VERSION}"
+KUBEADM_FLAGS+=" --apiserver-advertise-address ${PUBLICIP}"
+KUBEADM_FLAGS+=" --apiserver-cert-extra-sans ${PUBLICIP} ${PRIVATEIP}"
+KUBEADM_FLAGS+=" --service-cidr ${SERVICE_CIDR}"
+
+if [[ "${CNI_PROVIDER}" == "calico" ]]; then
+  KUBEADM_FLAGS+=" --pod-network-cidr=${POD_CIDR}"
+fi
+
+kubeadm init ${KUBEADM_FLAGS}
+
+if [[ "${CNI_PROVIDER}" == "calico" ]]; then
+	# install calico
+	kubectl apply --kubeconfig /etc/kubernetes/admin.conf -f /tmp/calico.yaml
+elif [[ "${CNI_PROVIDER}" == "weavenet" ]]; then
+	# install weavenet
+	sysctl net.bridge.bridge-nf-call-iptables=1
+	export kubever=$(kubectl version --kubeconfig /etc/kubernetes/admin.conf | base64 | tr -d '\n')
+	kubectl apply --kubeconfig /etc/kubernetes/admin.conf -f "https://cloud.weave.works/k8s/net?env.CHECKPOINT_DISABLE=1&env.IPALLOC_RANGE=${POD_CIDR}&disable-npc=true&k8s-version=$kubever"
+else
+	echo "CNI_PROVIDER:${CNI_PROVIDER} not in ['calico', 'weavenet']"
+	echo "Failed to install pod network."
+	exit 2
```

Comment:
exit code 2 represents "Misuse of shell builtins". Just do `exit 1` please unless you have a good reason for 2.